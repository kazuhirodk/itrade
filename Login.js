import React, {Component} from 'react'
import {View, Text, StyleSheet, Image,
        TouchableWithoutFeedback, StatusBar, TextInput,
        SafeAreaView, Keyboard, TouchableOpacity,
        KeyboardAvoidingView} from 'react-native'

export default class Login extends Component {
    render(){
        return (
            <SafeAreaView style={styles.container}>
                <StatusBar barStyle="light-content"/>
                <KeyboardAvoidingView behavior='padding' style={styles.container}>
                    <TouchableWithoutFeedback style={styles.container} onPress={Keyboard.dismiss}>
                        <View style = {styles.logoContainer}>
                            <View style={styles.logoContainer}>
                                <Text style={styles.title}>iTrade</Text>
                                <Text style={styles.rodape}>Registrar-se</Text>
                            </View>
                            
                        </View>
                    </TouchableWithoutFeedback>  
                    <View style={styles.infoContainer}>
                                <TextInput style={styles.input}
                                    placeholder="Nome de usuÃ¡rio"
                                    placeholderTextColor="rgba(255,255,255,0.8)"
                                    returnKeyType='next'
                                    autoCorrect ={false}
                                    onSubmitEditing={()=> this.refs.txtPassword.focus()}
                                />
                                <TextInput style={styles.input}
                                    placeholder="Senha"
                                    placeholderTextColor="rgba(255,255,255,0.8)"
                                    returnKeyType='go'
                                    autoCorrect ={false}
                                    secureTextEntry={true}
                                    ref={"txtPassword"}
                                />
                                <TouchableOpacity style={styles.buttonContainer}>
                                    <Text style={styles.buttonText}>LOGIN</Text>
                                </TouchableOpacity>    
                            </View>
                </KeyboardAvoidingView>
            </SafeAreaView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'rgb(32, 53, 70)',
        flexDirection: 'column', 
        
    },
    logoContainer: {
       alignItems: 'center',
       justifyContent: 'center',
       flex:1
     
    },
   
    title:{
        fontWeight: 'bold',
        fontSize: 28,
        color: 'white', 
    },
    rodape: {
       fontWeight: 'bold',
       fontSize: 18,
       color: '#f7c744',
       textAlign: 'center', 
       marginTop: 5,
       opacity: 0.9,
       
    },
    infoContainer: {
        position:'absolute',
        left: 0,
        right: 0,
        bottom: 0,
        height: 200,
        padding: 20,
        //backgroundColor: 'red'
    },
    input: {
        height: 40,
        backgroundColor:'rgba(255,255,255,0.2)',
        color: "#FFF",
        marginBottom: 20,
        paddingHorizontal: 10
    },

    buttonContainer: {
        backgroundColor: '#f7c744',
        paddingVertical: 15
    },
    buttonText: {
        textAlign:'center',
        color:'black',
        fontWeight: "bold",
        fontSize: 18
        
    }

    
})