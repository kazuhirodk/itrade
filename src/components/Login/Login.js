import React, { Component } from 'react';
import { Alert, View, Image, StyleSheet, KeyboardAvoidingView, Text, TextInput, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';
import firebase from 'react-native-firebase';

export default class Login extends React.Component {
  goToSignUp = () => {
    Actions.signup()
  }
  goToHome = () => {
    Actions.home()
  }

  state = {
    email: '',
    password: '',
    isAuthenticated: false,
  }

  componentDidMount() {
    firebase.auth().onAuthStateChanged(user => {
      if(user)
        this.goToHome()    
    })
  }

  login = async () => {
    const { email, password } = this.state;

    try {
      const user = await firebase.auth()
        .signInWithEmailAndPassword(email, password);

      //this.setState({ isAuthenticated: true });
      const { currentUser } = firebase.auth()
      this.setState({ currentUser })
      this.goToHome();
      //console.log(user);
    } catch (err) {
      Alert.alert('Usuário ou senha inválida')
      console.log(err);
    }
  }

  render() {
    return (
      <KeyboardAvoidingView style={styles.container}>
        <View style={styles.loginContainer}>
          <Image resizeMode="contain" style={styles.logo} source={require('../../components/images/itrade-logo.png')} />
        </View>

        <View style={styles.formContainer}>
          <TextInput style = {styles.input}
            value={this.state.email}
            onChangeText={email => this.setState({ email })}
            autoCapitalize="none"
            onSubmitEditing={() => this.passwordInput.focus()}
            autoCorrect={false}
            keyboardType='email-address'
            returnKeyType="next"
            placeholder='Email'
            placeholderTextColor='#CD7F32'/>

          <TextInput style = {styles.input}
            value={this.state.password}
            onChangeText={password => this.setState({ password })}
            returnKeyType="go"
            placeholder='Senha'
            placeholderTextColor='#CD7F32'
            secureTextEntry/>

          <TouchableOpacity style={styles.buttonContainer} onPress= {this.login} >
            <Text style={styles.buttonText}>LOGIN</Text>
          </TouchableOpacity>

          <TouchableOpacity style={styles.signupButton} onPress = {this.goToSignUp} >
            <Text style={styles.signupButton}>Registre-se</Text>
          </TouchableOpacity>
        </View>
      </KeyboardAvoidingView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 20,
    flex: 1,
    backgroundColor: '#FFFAFA'
  },
  loginContainer:{
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center'
  },
  logo: {
    position: 'absolute',
    width: 300,
    height: 100
  },
  input:{
    height: 40,
    backgroundColor: '#E3E1E1',
    marginBottom: 10,
    padding: 10,
    color: '#000'
  },
  buttonContainer:{
    backgroundColor: '#FD6D64',
    paddingVertical: 15,
    marginBottom: 10
  },
  buttonText:{
    color: '#fff',
    textAlign: 'center',
    fontWeight: '700'
  },
  signupButton:{
    marginBottom: 20,
    textAlign: 'center',
    color: '#CD7F32'
  }
})
