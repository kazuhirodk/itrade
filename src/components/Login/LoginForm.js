import React, { Component } from 'react';
import { View, Text, TextInput, TouchableOpacity, Alert, Button, StyleSheet, StatusBar } from 'react-native';

class LoginForm extends Component {
  render() {
    return (
      <View style={StyleSheet.container}>
        <TextInput style = {styles.input}
          autoCapitalize="none"
          onSubmitEditing={() => this.passwordInput.focus()}
          autoCorrect={false}
          keyboardType='email-address'
          returnKeyType="next"
          placeholder='Email'
          placeholderTextColor='#CD7F32'/>

        <TextInput style = {styles.input}
          returnKeyType="go"
          ref={(input)=> this.passwordInput = input}
          placeholder='Senha'
          placeholderTextColor='#CD7F32'
          secureTextEntry/>

        <TouchableOpacity style={styles.buttonContainer}>
          <Text  style={styles.buttonText}>LOGIN</Text>
        </TouchableOpacity>

        <Text style={styles.subscribeText}>Registre-se</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 20
  },
  input:{
    height: 40,
    backgroundColor: '#E3E1E1',
    marginBottom: 10,
    padding: 10,
    color: '#000'
  },
  buttonContainer:{
    backgroundColor: '#FD6D64',
    paddingVertical: 15,
    marginBottom: 10
  },
  buttonText:{
    color: '#fff',
    textAlign: 'center',
    fontWeight: '700'
  },
  subscribeText:{
    marginBottom: 30,
    alignSelf: 'center',
    color: '#CD7F32',
    fontWeight: '700'
  }
});

export default LoginForm;
