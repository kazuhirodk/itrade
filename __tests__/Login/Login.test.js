import Login from '../../src/components/Login/Login.js'
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

describe('#Login', () => {
  it('matches snapshot properly', () => {
    const jsonLogin = renderer.create(<Login />).toJSON();
    expect(jsonLogin).toMatchSnapshot();
  });
});
