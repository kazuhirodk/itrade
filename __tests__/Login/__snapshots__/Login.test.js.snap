// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Login matchs snapshot properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFAFA",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/components/images/itrade-logo.png",
        }
      }
      style={
        Object {
          "height": 100,
          "position": "absolute",
          "width": 300,
        }
      }
    />
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      placeholderTextColor="#CD7F32"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#E3E1E1",
          "color": "#000",
          "height": 40,
          "marginBottom": 10,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Senha"
      placeholderTextColor="#CD7F32"
      rejectResponderTermination={true}
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#E3E1E1",
          "color": "#000",
          "height": 40,
          "marginBottom": 10,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FD6D64",
          "marginBottom": 10,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        LOGIN
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "color": "#CD7F32",
          "marginBottom": 20,
          "opacity": 1,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#CD7F32",
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Registre-se
      </Text>
    </View>
  </View>
</View>
`;
